dipy/workflows/base.py

```diff
@@ -60,18 +60,39 @@ def __init__(self, prog=None, usage=None, description=None, epilog=None,
         self.doc = None
 
     def add_workflow(self, workflow):
-        specs = inspect.getargspec(workflow)
-        doc = inspect.getdoc(workflow)
-        self.doc = NumpyDocString(doc)['Parameters']
+        """ Take a workflow object and use introspection to extract the parameters,
+        types and docstrings of its run method. Then add these parameters
+        to the current arparser's own params to parse. If the workflow is of
+        type combined_workflow, the optional input parameters of its
+        sub workflows will also be added.
+
+        Parameters
+        -----------
+        workflow : dipy.workflows.workflow.Workflow
+            Workflow from which to infer parameters.
+
+        Returns
+        -------
+        sub_flow_optionals : dictionary of all sub workflow optional parameters
+        """
+        specs = inspect.getargspec(workflow.run)
```

Comment:
@matthieudumont I think as mentioned in the documentation using inspect.signature may solve the problem.